crates/uv-interpreter/src/interpreter.rs

```diff
@@ -346,58 +361,22 @@ struct InterpreterInfo {
 impl InterpreterInfo {
     /// Return the resolved [`InterpreterInfo`] for the given Python executable.
     pub(crate) fn query(interpreter: &Path) -> Result<Self, Error> {
-        let script = include_str!("get_interpreter_info.py");
-        let output = if cfg!(windows)
-            && interpreter
-                .extension()
-                .is_some_and(|extension| extension == "bat")
-        {
-            // Multiline arguments aren't well-supported in batch files and `pyenv-win`, for example, trips over it.
-            // We work around this batch limitation by passing the script via stdin instead.
-            // This is somewhat more expensive because we have to spawn a new thread to write the
-            // stdin to avoid deadlocks in case the child process waits for the parent to read stdout.
-            // The performance overhead is the reason why we only applies this to batch files.
-            // https://github.com/pyenv-win/pyenv-win/issues/589
-            let mut child = Command::new(interpreter)
-                .arg("-")
-                .stdin(std::process::Stdio::piped())
-                .stdout(std::process::Stdio::piped())
-                .spawn()
-                .map_err(|err| Error::PythonSubcommandLaunch {
-                    interpreter: interpreter.to_path_buf(),
-                    err,
-                })?;
-
-            let mut stdin = child.stdin.take().unwrap();
-
-            // From the Rust documentation:
-            // If the child process fills its stdout buffer, it may end up
-            // waiting until the parent reads the stdout, and not be able to
-            // read stdin in the meantime, causing a deadlock.
-            // Writing from another thread ensures that stdout is being read
-            // at the same time, avoiding the problem.
-            std::thread::spawn(move || {
-                stdin
-                    .write_all(script.as_bytes())
-                    .expect("failed to write to stdin");
-            });
-
-            child.wait_with_output()
-        } else {
-            Command::new(interpreter).arg("-c").arg(script).output()
-        }
-        .map_err(|err| Error::PythonSubcommandLaunch {
-            interpreter: interpreter.to_path_buf(),
-            err,
-        })?;
+        let tempdir = tempdir()?;
```

Comment:
Not to be annoying here, but are we going to say we're okay with temporary directories outside of scopes we control?

Reply:
Unless there is a specific problem with the system directories, i believe we should follow system conventions and user's settings over rolling our own. For our own garbage collection, we'd also have to implement our own synchronization when there are multiple uv instances running in parallel.

Reply:
I'd say let's just put it in the cache for consistency. It shouldn't be too hard here, right?

Reply:
What synchronization would we need? I would just delete old temporary directories (i.e. 24 hours) similar to how `/tmp` can be cleaned at arbitrary times.

Reply:
We can do that. What does it give us over the system temp?

Reply:
I also don't follow the motivation yet.

Reply:
I've heard of problems being caused by the system cleaning up temporary directories at arbitrary times which can break your program. Notably though we've already had problems with the system temporary directory (as discussed in the linking issue) and it seems safer to just use a directory we control. I'm totally fine with us deciding we want to use the system temporary directory, but I want us to be consistent if feasible or be clear about _when_ we should use one vs another.

@charliermarsh previously @konstin said that using the system temporary directory was preferred for automated cleanup if we fail to clean up the directory i.e. if we crash.

Reply:
> I've heard of problems being caused by the system cleaning up temporary directories at arbitrary times which can break your program. 

If we can get a reproduction for that i'm all for switching to never using system temp again.

> Notably though we've already had problems with the system temporary directory (as discussed in the copy-on-write issue) and it seems safer to just use a directory we control. 

That is a very specific problem: Operating system don't offer atomic write operations, so e.g. two installer threads writing the same file could produce something broken/crash. Renames are (assumed to be) atomic and replace the existing file. Renames don't work across device boundaries (it would need a copy), so we need to create a named tempfile/tempdir in the same location as the final file, write to it and persist it. Some people e.g. put a ramdisk as their `/tmp` or the target fs is part of an overlayfs while tmp isn't, or the project lies on a different partition or drive than the system.